engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnVariableMappingBehavior.java

```diff
@@ -105,7 +102,21 @@ public BpmnVariableMappingBehavior(
                     scopeKey, processDefinitionKey, processInstanceKey, result);
                 return null;
               });
+
+    } else if (temporaryVariables != null) {
+      // merge/propagate the event variables by default
+      variableBehavior.mergeDocument(
+          elementInstanceKey, processDefinitionKey, processInstanceKey, temporaryVariables);
+
+      variablesState.removeTemporaryVariables(elementInstanceKey);
+
+    } else if (isConnectedToEventBasedGateway(element)) {
+      // event variables are set local variables instead of temporary variables
+      final var localVariables = variablesState.getVariablesLocalAsDocument(elementInstanceKey);
+      variableBehavior.mergeDocument(
+          scopeKey, processDefinitionKey, processInstanceKey, localVariables);
     }
```

Comment:
There's an argument to make that merging of event variables is not part output mappings, but I don't mind it here to be honest.

Reply:
I would like to have it here because it behaves the same as for other elements. For example, the message variables of an intermediate catch event are first stored as temporary variables. If there are no output mappings then the variables are propagated (line 108).